packages/core/src/external/monitoring/heartbeat.ts

```diff
@@ -5,7 +5,12 @@ import axios from "axios";
  * Notifies our monitoring service that the service ran successfully.
  */
 export async function sendHeartbeatToMonitoringService(url: string): Promise<void> {
-  await executeWithNetworkRetries(async () => {
-    await axios.post(url);
-  });
+  await executeWithNetworkRetries(
+    async () => {
+      await axios.post(url);
+    },
+    {
+      httpStatusCodesToRetry: [500, 502, 503, 504],
```

Comment:
429 too

I'd use `...defaultOptionsRequestNotAccepted` and add the codes you want.